Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -395,13 +444,188 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
                ge_iff_le, forall_exists_index]
     refine ⟨0, fun a i hia ↦ le_trans (integral_nonneg f_nn) (hia i rfl.le)⟩
 
-/-- The topology of the Lévy-Prokhorov metric is finer than the topology of convergence in
+/-- The topology of the Lévy-Prokhorov metric is at least as fine as the topology of convergence in
 distribution. -/
 theorem levyProkhorov_le_convergenceInDistribution :
     TopologicalSpace.coinduced (LevyProkhorov.probabilityMeasure (Ω := Ω)) inferInstance
       ≤ (inferInstance : TopologicalSpace (ProbabilityMeasure Ω)) :=
   (continuous_levyProkhorov_to_probabilityMeasure).coinduced_le
 
-end Levy_Prokhorov_comparison
+end Levy_Prokhorov_is_finer
+
+
+
+section Levy_Prokhorov_metrizes_convergence_in_distribution
+
+/-! ### On separable spaces the Lévy-Prokhorov distance metrizes convergence in distribution -/
+
+open BoundedContinuousFunction TopologicalSpace
+
+variable {ι : Type*} {Ω : Type*} [MeasurableSpace Ω]
+
+/-- In a separable pseudometric space, for any ε > 0 there exists a countable collection of
+disjoint Borel measurable subsets of diameter at most ε that cover the whole space. -/
+lemma SeparableSpace.exists_measurable_partition_diam_le
+    (X : Type*) [PseudoMetricSpace X] [SeparableSpace X]
+    [MeasurableSpace X] [OpensMeasurableSpace X] {ε : ℝ} (ε_pos : 0 < ε) :
+    ∃ (As : ℕ → Set X), (∀ n, MeasurableSet (As n)) ∧ (∀ n, Bornology.IsBounded (As n)) ∧
+        (∀ n, diam (As n) ≤ ε) ∧ (⋃ n, As n = univ) ∧
+        (Pairwise (fun (n m : ℕ) ↦ Disjoint (As n) (As m))) := by
+  by_cases X_emp : IsEmpty X
+  · refine ⟨fun _ ↦ ∅, fun _ ↦ MeasurableSet.empty, fun _ ↦ Bornology.isBounded_empty, ?_, ?_,
+            fun _ _ _ ↦ disjoint_of_subsingleton⟩
+    · intro n
+      simpa only [diam_empty] using LT.lt.le ε_pos
+    · simp only [iUnion_empty]
+      apply Eq.symm
+      simp only [univ_eq_empty_iff, X_emp]
+  rw [not_isEmpty_iff] at X_emp
+  obtain ⟨xs, xs_dense⟩ := exists_dense_seq X
+  have half_ε_pos : 0 < ε / 2 := half_pos ε_pos
+  set Bs := fun n ↦ Metric.ball (xs n) (ε/2)
+  set As := disjointed Bs
+  refine ⟨As, ?_, ?_, ?_, ?_, ?_⟩
+  · exact MeasurableSet.disjointed (fun n ↦ measurableSet_ball)
+  · exact fun n ↦ Bornology.IsBounded.subset isBounded_ball <| disjointed_subset Bs n
+  · intro n
+    apply (diam_mono (disjointed_subset Bs n) isBounded_ball).trans
+    convert diam_ball half_ε_pos.le
+    ring
+  · have aux : ⋃ n, Bs n = univ := by
+      convert DenseRange.iUnion_uniformity_ball xs_dense <| Metric.dist_mem_uniformity half_ε_pos
+      exact (ball_eq_ball' _ _).symm
+    simpa only [← aux] using iUnion_disjointed
+  · exact disjoint_disjointed Bs
+
+variable [PseudoMetricSpace Ω] [OpensMeasurableSpace Ω]
```

Comment:
Since you are using `[SeparableSpace Ω]` in all the statements, maybe factor it out to the `variable` line? Maybe even move it before the previous lemma, so that it can take advantage of it?